resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/submit/DriverInitContainerComponentsProvider.scala

```diff
@@ -128,28 +129,36 @@ private[spark] class DriverInitContainerComponentsProviderImpl(
           maybeResourceStagingServerInternalTrustStorePassword,
           maybeResourceStagingServerInternalTrustStoreType,
           INIT_CONTAINER_SECRET_VOLUME_MOUNT_PATH)
-    }
-    new SparkInitContainerConfigMapBuilderImpl(
+      }
+      new SparkInitContainerConfigMapBuilderImpl(
         sparkJars,
         sparkFiles,
         jarsDownloadPath,
         filesDownloadPath,
         configMapName,
         configMapKey,
         submittedDependencyConfigPlugin)
+    } else new SparkInitContainerEmptyConfigMapBuilderImpl()
   }
 
   override def provideContainerLocalizedFilesResolver(): ContainerLocalizedFilesResolver = {
     new ContainerLocalizedFilesResolverImpl(
         sparkJars, sparkFiles, jarsDownloadPath, filesDownloadPath)
   }
+  val containerLocalizedFilesResolver = provideContainerLocalizedFilesResolver()
+  val resolvedSparkJars = containerLocalizedFilesResolver.resolveSubmittedSparkJars()
+  val resolvedSparkFiles = containerLocalizedFilesResolver.resolveSubmittedSparkFiles()
+  // Bypass init-containers if `spark.jars` and `spark.files` is empty or only has `local://` URIs
+  private val needInitContainer = (resolvedSparkJars ++ resolvedSparkFiles).exists { e =>
+    e.nonEmpty && !e.startsWith("local://")
```

Comment:
Try to refrain from this check by leveraging the functions already provided by: [here
](https://github.com/apache-spark-on-k8s/spark/blob/branch-2.1-kubernetes/resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/submit/KubernetesFileUtils.scala) or if use-case is specific enough try adding the function to the Utils for re-usability within the system